src/main/java/dev/bitbite/logging/log/ConsoleLog.java

```diff
@@ -24,22 +25,18 @@ public ConsoleLog() {
 	}
 
 	@Override
-	public void log(LogLevel logLevel, Category category, String message) {
-		System.out.println(this.format(logLevel, category, message));
-	}
-	
-	@Override
-	public void log(LogLevel logLevel, String message) {
-		System.out.println(this.format(logLevel, message));
-	}
-
-	@Override
-	public void log(LogLevel logLevel, Category category, Exception exception) {
-		System.out.println(this.format(logLevel, category, exception.getMessage()));
-	}
-	
-	@Override
-	public void log(LogLevel logLevel, Exception exception) {
-		System.out.println(this.format(logLevel, exception.getMessage()));
+	public void log(LogMessage logMessage) {
+		String logContent = "";
+		if(logMessage.message != null) {
+			logContent += logMessage.message;
+		}
+		if(logMessage.message != null && logMessage.exception != null) {
+			logContent += " ";
+		}
+		if(logMessage.exception != null) {
+			logContent += logMessage.exception.getMessage()+" in "+logMessage.exception.getStackTrace()[0].getClassName()+":"+logMessage.exception.getStackTrace()[0].getLineNumber();
+		}
+		System.out.println(this.format(logMessage.logLevel, logMessage.category, logContent));
+		if(logMessage.exception != null) Arrays.stream(logMessage.exception.getStackTrace()).skip(1).forEach(e -> { System.out.println(this.format(LogLevels.STACKTRACE, logMessage.category, e.getClassName()+" in line "+e.getLineNumber())); });
```

Comment:
Please add linebreaks to improve readability

Reply:
should be done, @see latestCommit